account_bank_statement_import_camt/camt.py

```diff
@@ -118,27 +124,54 @@ def parse_transaction(self, ns, node):
             ns, node, './ns:BookgDt/ns:Dt', transaction, 'execution_date')
         self.add_value_from_node(
             ns, node, './ns:ValDt/ns:Dt', transaction, 'value_date')
+        amount = self.parse_amount(ns, node)
+        if amount != 0.0:
+            transaction['amount'] = amount
+        self.add_value_from_node(
+            ns, node, './ns:AddtlNtryInf', transaction, 'name')
+        self.add_value_from_node(
+            ns, node, [
+                './ns:NtryDtls/ns:RmtInf/ns:Strd/ns:CdtrRefInf/ns:Ref',
+                './ns:NtryDtls/ns:Btch/ns:PmtInfId',
+            ],
+            transaction, 'ref'
+        )
 
-        transaction['amount'] = self.parse_amount(ns, node)
-
-        details_node = node.xpath(
+        details_nodes = node.xpath(
             './ns:NtryDtls/ns:TxDtls', namespaces={'ns': ns})
-        if details_node:
-            self.parse_transaction_details(ns, details_node[0], transaction)
-        if not transaction.get('name'):
-            self.add_value_from_node(
-                ns, node, './ns:AddtlNtryInf', transaction, 'name')
-        if not transaction.get('name'):
-            transaction['name'] = '/'
-        if not transaction.get('ref'):
-            self.add_value_from_node(
-                ns, node, [
-                    './ns:NtryDtls/ns:Btch/ns:PmtInfId',
-                ],
-                transaction, 'ref'
-            )
-        transaction['data'] = etree.tostring(node)
-        return transaction
+        if len(details_nodes) == 0:
+            yield transaction
+            return
+        transaction_base = transaction
+        for i, dnode in enumerate(details_nodes):
+            transaction = transaction_base.copy()
+            self.parse_transaction_details(ns, dnode, transaction)
+            # transactions['data'] should be a synthetic xml snippet which
+            # contains only the TxDtls that's relevant.
```

Comment:
Wasted effort maybe: that field isn't super important afaik and #103 is trying to remove the field anyway. I'd be ok with retracting that commit.